src/Nethermind/Nethermind.Core/Threading/ZeroContentionCounter.cs

```diff
@@ -0,0 +1,94 @@
+using System.Threading;
+
+namespace Nethermind.Core.Threading;
+public class ZeroContentionCounter
+{
+    private ThreadLocal<BoxedLong> _threadLocal = new(() => new BoxedLong(), trackAllValues: true);
+
+    private static Func<ThreadLocal<BoxedLong>, long> _totalDelegate = CreateTotalDelegate();
+
+    public long GetTotalValue()
+    {
+        return _totalDelegate(_threadLocal);
+    }
+
+    private static Func<ThreadLocal<BoxedLong>, long> CreateTotalDelegate()
+    {
+        FieldInfo linkedSlot = typeof(ThreadLocal<BoxedLong>).GetField("_linkedSlot", BindingFlags.NonPublic | BindingFlags.Instance)!;
+        FieldInfo next = linkedSlot.FieldType.GetField("_next", BindingFlags.NonPublic | BindingFlags.Instance)!;
+        FieldInfo value = next.FieldType.GetField("_value", BindingFlags.NonPublic | BindingFlags.Instance)!;
+
+        // Parameters
+        ParameterExpression threadLocalParam = Expression.Parameter(typeof(ThreadLocal<BoxedLong>), "threadLocal");
+
+        // Fields
+        MemberExpression linkedSlotField = Expression.Field(threadLocalParam, linkedSlot);
+
+        // Variables
+        ParameterExpression linkedSlotVar = Expression.Variable(linkedSlotField.Type, "linkedSlot");
+        ParameterExpression totalVar = Expression.Variable(typeof(long), "total");
+
+        // Assignments
+        BinaryExpression assignLinkedSlot = Expression.Assign(linkedSlotVar, linkedSlotField);
+        BinaryExpression assignTotal = Expression.Assign(totalVar, Expression.Constant(0L));
+        BinaryExpression assignNextSlot = Expression.Assign(linkedSlotVar, Expression.Field(linkedSlotVar, next));
+
+        // Labels
+        LabelTarget breakLabel = Expression.Label(typeof(long), "breakLabel");
+
+        ConditionalExpression breakCondition =
+            Expression.IfThen(
+                Expression.Equal(linkedSlotVar, Expression.Constant(null)),
+                Expression.Break(breakLabel, totalVar)
+            );
+
+        // Loop body
+        BlockExpression loopBody = Expression.Block(
+            breakCondition,
+            Expression.AddAssign(
+                totalVar,
+                Expression.Property(
+                    Expression.Field(linkedSlotVar, value), typeof(BoxedLong),
+                    nameof(BoxedLong.Value)
+                )
+            ),
+            assignNextSlot
+        );
+
+        // Loop
+        LoopExpression loop = Expression.Loop(loopBody);
+
+        // Block
+        BlockExpression block = Expression.Block(
+            new[] { linkedSlotVar, totalVar },
+            assignLinkedSlot,
+            assignTotal,
+            breakCondition,
+            assignNextSlot,
+            loop,
+            Expression.Label(breakLabel, totalVar)
+        );
+
+        // Lambda
+        Expression<Func<ThreadLocal<BoxedLong>, long>> lambda =
+            Expression.Lambda<Func<ThreadLocal<BoxedLong>, long>>(block, threadLocalParam);
+
+        return lambda.Compile();
```

Comment:
Can you add equivalent pseudo-code for that in comment?

Reply:
Added